spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -175,7 +181,25 @@ public Statistics estimateStatistics() {
   protected Statistics estimateStatistics(Snapshot snapshot) {
     // its a fresh table, no data
     if (snapshot == null) {
-      return new Stats(0L, 0L);
+      return new Stats(0L, 0L, Maps.newHashMap());
+    }
+
+    Map<NamedReference, ColumnStatistics> map = Maps.newHashMap();
+
+    if (readConf.enableColumnStats()) {
+      List<StatisticsFile> files = table.statisticsFiles();
+      if (!files.isEmpty()) {
+        List<BlobMetadata> metadataList = (files.get(0)).blobMetadata();
+
+        for (BlobMetadata blobMetadata : metadataList) {
+          long ndv = Long.parseLong(blobMetadata.properties().get("ndv"));
```

Comment:
I think we should be more defensive if there is no value, or numberformatexception?

Reply:
Also check `org.apache.iceberg.BlobMetadata#type`?

Reply:
I have added the code to check null and empty value. It's probably too strict to throw Exception if the value is null or empty String. I log a DEBUG message instead.

Is there an agreed type for `org.apache.iceberg.BlobMetadata#type`? If yes, I will add a check.

Reply:
`org.apache.iceberg.puffin.StandardBlobTypes` has the supported types, but currently its just one